src/screens/AddWalletSheet.tsx

```diff
@@ -0,0 +1,283 @@
+import { AddWalletList } from '@/components/add-wallet/AddWalletList';
+import { AddWalletItem } from '@/components/add-wallet/AddWalletRow';
+import { Box, globalColors, Inset, Stack, Text } from '@/design-system';
+import { useNavigation } from '@/navigation';
+import Routes from '@/navigation/routesNames';
+import React, { useCallback, useRef } from 'react';
+import * as i18n from '@/languages';
+import { HARDWARE_WALLETS, PROFILES, useExperimentalFlag } from '@/config';
+import { analytics, analyticsV2 } from '@/analytics';
+import { InteractionManager } from 'react-native';
+import { createAccountForWallet, walletsLoadState } from '@/redux/wallets';
+import WalletBackupTypes from '@/helpers/walletBackupTypes';
+import { createWallet } from '@/model/wallet';
+import WalletTypes from '@/helpers/walletTypes';
+import logger from '@/utils/logger';
+import { captureException } from '@sentry/react-native';
+import { useDispatch } from 'react-redux';
+import { backupUserDataIntoCloud } from '../handlers/cloudBackup';
+import showWalletErrorAlert from '@/helpers/support';
+import { WalletLoadingStates } from '@/helpers/walletLoadingStates';
+import { useWallets } from '@/hooks';
+
+const TRANSLATIONS = i18n.l.wallet.new.add_wallet;
+
+export const AddWalletSheet = () => {
+  const { goBack, navigate } = useNavigation();
+
+  const hardwareWalletsEnabled = useExperimentalFlag(HARDWARE_WALLETS);
+  const profilesEnabled = useExperimentalFlag(PROFILES);
+  const dispatch = useDispatch();
+  const creatingWallet = useRef<boolean>();
+  const {
+    isDamaged,
+    selectedWallet,
+    setIsWalletLoading,
+    wallets,
+  } = useWallets();
+
+  const onPressCreate = useCallback(async () => {
```

Comment:
Not sure it is worth using callbacks here at all. We are not memoizing the objects created below them, maybe it makes sense to completely ditch the `useCallback` here. It could actually be more performant than with it, due to less memoization overhead